packages/core/content-manager/server/src/history/services/history.ts

```diff
@@ -59,6 +68,61 @@ const createHistoryService = ({ strapi }: { strapi: LoadedStrapi }) => {
     return documentMetadataService.getStatus(document, meta.availableStatus);
   };
 
+  /**
+   * Creates a populate object that looks for all the relations that need
+   * to be saved in history, and populates only the documentId and locale for each of them.
+   */
+  const getDeepPopulate = (uid: Common.UID.Schema) => {
+    const model = strapi.getModel(uid);
+    const attributes = Object.entries(model.attributes);
+
+    return attributes.reduce((acc: any, [attributeName, attribute]) => {
+      switch (attribute.type) {
+        case 'relation': {
+          if (
+            ['createdBy', 'updatedBy', 'strapi_stage', 'strapi_assignee'].includes(attributeName)
+          ) {
+            break;
+          }
+          const isVisible = contentTypes.isVisibleAttribute(model, attributeName);
```

Comment:
those should not be visible attributes so they are going to be excluded by the isVisible check